packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeDefinition.cs

```diff
@@ -1036,7 +1060,7 @@ private static MethodBodyStatement DeserializationPropertyNullCheckStatement(
             CSharpType serializedType = property.Type;
             var propertyIsRequired = property.WireInfo?.IsRequired ?? false;
 
-            if (serializedType.IsNullable)
+            if (property.WireInfo!.IsNullable)
```

Comment:
I think it is more correct to use the IsNullable value from the spec as opposed to the CSharpType. These can differ for customized properties.

Reply:
Actually, I think we need to use IsRequired from the spec to handle the case where IsNullable is set to true on the CSharpType for non-required properties, but still use CSharpType.IsNullable for the case that the type is itself nullable.